src/test/java/com/conveyal/gtfs/validator/FlexValidatorTest.java

```diff
@@ -100,192 +141,285 @@ private static Stream<LocationArguments> createLocationChecks() {
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_STOP_ID)
             ),
             new BaseArguments(
-                createStopTime(1130, INT_MISSING, 1100, INT_MISSING),
-                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_ARRIVAL_TIME, NewGTFSErrorType.FLEX_FORBIDDEN_START_PICKUP_DROP_OFF_WINDOW)
+                createStopTimeForStopIdTest("stop-id-1", null, "location-id-1"),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_STOP_ID)
+            )
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("createLocationGroupIdTests")
+    void validateLocationGroupId(BaseArguments baseArguments) {
+        List<NewGTFSError> errors = new ArrayList<>();
+        FlexValidator.validateLocationGroupId((StopTime) baseArguments.testObject, errors);
+        checkValidationErrorsMatchExpectedErrors(errors, baseArguments.expectedErrors);
+    }
+
+    private static Stream<BaseArguments> createLocationGroupIdTests() {
+        return Stream.of(
+            new BaseArguments(
+                createStopTimeForStopIdTest(null, "location-group-id-1", null),null
             ),
             new BaseArguments(
-                createStopTime(1130, INT_MISSING, INT_MISSING, 1200),
-                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_ARRIVAL_TIME, NewGTFSErrorType.FLEX_FORBIDDEN_END_PICKUP_DROP_OFF_WINDOW)
+                createStopTimeForStopIdTest(null, null, null),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_LOCATION_GROUP_ID)
             ),
             new BaseArguments(
-                createStopTime(INT_MISSING, 1330, 1100, INT_MISSING),
-                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_DEPARTURE_TIME, NewGTFSErrorType.FLEX_FORBIDDEN_START_PICKUP_DROP_OFF_WINDOW)
+                createStopTimeForStopIdTest("stop-id-1", "location-group-id-1", null),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_LOCATION_GROUP_ID)
             ),
             new BaseArguments(
-                createStopTime(INT_MISSING, 1330, INT_MISSING, 1200),
-                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_DEPARTURE_TIME, NewGTFSErrorType.FLEX_FORBIDDEN_END_PICKUP_DROP_OFF_WINDOW)
+                createStopTimeForStopIdTest(null, "location-group-id-1", "location-id-1"),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_LOCATION_GROUP_ID)
             )
         );
     }
 
     @ParameterizedTest
-    @MethodSource("createStopTimeChecksForStartEndPickupDropoff")
-    void validateStopTimeStartEndPickupDropoffTests(StopTimeArguments stopTimeArguments
-    ) {
-        List<NewGTFSError> errors = FlexValidator.validateStopTime(
-            (StopTime) stopTimeArguments.testObject,
-            stopTimeArguments.stopAreas,
-            stopTimeArguments.locations
+    @MethodSource("createLocationIdTests")
+    void validateLocationId(BaseArguments baseArguments) {
```

Comment:
You could spread `baseArguments` and put its attributes as this method's arguments instead.